### PR TITLE
EVA-2158 Add dependency between bsub commands so they are run in series

### PR DESCRIPTION
**Pull request by andresfsilva**

### CODE REVIEW

File: eva-accession-clustering-automation/cluster_multiple_assemblies.py [hunk 1]
```diff
@@ -44,6 +47,10 @@ def generate_bsub_command(assembly_accession, properties_path, clustering_artifa
     return command
 
 
+def get_job_name(assembly_accession):
+    return 'cluster_{assembly_accession}'.format(assembly_accession=assembly_accession)
```
Comment (jmmut) [inline eva-accession-clustering-automation/cluster_multiple_assemblies.py:51]: we have the option here of putting a timestamp in the job name. That way, if a run fails, the next run will be using a different set of dependencies.

In the past we have had problems when submitting different jobs with the same name, if they were submitted too fast (right after the previous one failed).

File: eva-accession-clustering-automation/cluster_multiple_assemblies.py [hunk 2]
```diff
@@ -34,8 +34,11 @@ def generate_bsub_command(assembly_accession, properties_path, clustering_artifa
     if memory:
         memory_amount = memory
 
-    command = 'bsub -J {job_name} -o {log_file} -e {error_file} -M {memory_amount} -R "rusage[mem={memory_amount}]" ' \
-              'java -jar {clustering_artifact} --spring.config.location=file:{properties_path}'\
+    command = 'bsub '
+    if dependency:
+        command += '-w {dependency} '.format(dependency=dependency)
```
Comment (jmmut) [inline eva-accession-clustering-automation/cluster_multiple_assemblies.py:39]: you are mixing `+=` concatenation with newline concatenation (`' ' \\n ' '`) and  `.format`ting. I suggest a more consistent approach.

Also, not sure if you know you can do this:
```suggestion
        command += '-w {} '.format(dependency)
```
which may or may not be clearer, depending on the particular code.

**Review by jmmut — APPROVED** [review]
this looks like it works, but the changes I suggest are probably worth to do sooner rather than later.

File: eva-accession-clustering-automation/cluster_multiple_assemblies.py (hunk 1 above)
Comment (andresfsilva) [reply eva-accession-clustering-automation/cluster_multiple_assemblies.py]: Good point, I just changed it to include the timestamp in the job name

File: eva-accession-clustering-automation/cluster_multiple_assemblies.py (hunk 2 above)
Comment (andresfsilva) [reply eva-accession-clustering-automation/cluster_multiple_assemblies.py]: @jmmut Please let me know if you think the new approach looks better for you

**Review by tcezard — APPROVED** [review]

**Pull request merged**